tests/tensor/test_subtensor.py

```diff
@@ -105,13 +105,45 @@ def test_as_index_literal():
 
 
 class TestGetCanonicalFormSlice:
+    def test_none_constant(self):
+        slc, rev = get_canonical_form_slice(NoneConst, 5)
+        assert isinstance(slc, slice)
+        assert slc.start == 0
+        assert slc.stop == 5
+        assert slc.step == 1
+        assert rev == 1
+
     def test_scalar_constant(self):
         a = as_scalar(0)
         length = lscalar()
         res = get_canonical_form_slice(a, length)
         assert res[0].owner.op == ptb.switch
         assert res[1] == 1
 
+    def test_tensor_constant(self):
+        a = as_tensor(0)
```

Comment:
My understanding is this function should work also with `a = pt.lscalar("a")` but fails after the changes in this PR